mlflow/types/type_hints.py

```diff
@@ -39,6 +40,13 @@
 # special type hint that can be used to convert data to
 # the input example type after data validation
 TypeFromExample = TypeVar("TypeFromExample")
+OPTIONAL_INPUT_MSG = (
+    "Input cannot be Optional type. Fix this by removing the "
+    "Optional wrapper from the type hint. To use optional fields, "
+    "use a Pydantic-based type hint definition. See "
+    "https://docs.pydantic.dev/latest/api/base_model/ for pydantic "
```

Comment:
```suggestion
    "https://docs.pydantic.dev/latest/api/base_model/ for pydantic "
```

nit: Can we add a TODO to replace this with our doc that shows pydantic-based type hint?